### PR TITLE
More informative error messages for one, only

### PR DESCRIPTION
**Pull request by gabbard**
Closes erikrose/more-itertools#312

### CODE REVIEW

**Comment by gabbard** [comment]
I see some problems on CI; fixing them now

**Comment by gabbard** [comment]
@bbayles : CI is happy now, so this is ready for review.

File: more_itertools/more.py [hunk 1]
```diff
@@ -2522,9 +2525,9 @@ def time_limited(limit_seconds, iterable):
 
 def only(iterable, default=None, too_long=None):
     """If *iterable* has only one item, return it.
-    If it has zero items, return *default*.
-    If it has more than one item, raise the exception given by *too_long*,
-    which is ``ValueError`` by default.
+    If it has zero items, return *default*, which is *None* if not specified.
```
Comment (bbayles) [inline more_itertools/more.py]: Could you put this docstring back the way it was? Apologies, I am picky about this.

Comment (gabbard) [reply more_itertools/more.py]: @bbayles : Sure - can you clarify the scope of what you would like restored?  (the part about zero items, the part about if `too_long` is not specified, or both)?

Comment (bbayles) [reply more_itertools/more.py]: Both, if you could please.

Comment (gabbard) [reply more_itertools/more.py]: Restored!

**Pull request merged**